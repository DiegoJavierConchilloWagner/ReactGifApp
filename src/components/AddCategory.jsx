import React, { useState } from 'react'
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue( e.target.value );
    }

    const handleInputSubmit = (e) => {
        e.preventDefault();

        console.log('handleSubmit', inputValue);

        if( inputValue.trim().length > 2 ){
            setCategories( cats => [ inputValue, ...cats ]);
            setInputValue('')
        }

        
    }

    return (
        <form onSubmit={handleInputSubmit} >
            <p> { inputValue }</p>
            <input  
                type="text" 
                value={inputValue} 
                placeholder="Buscar meme..."
                onChange={ handleInputChange } 
            />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}