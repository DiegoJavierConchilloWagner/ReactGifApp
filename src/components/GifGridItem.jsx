import React from 'react'
import PropTypes from 'prop-types'


export const GifGridItem = ({ title, url }) => {

    return (
        <div className="gifCard animate__animated animate__bounceInUp">
            <img src={ url } alt={ title } />
            <p> { title } </p> 
        </div>

    )
}

GifGridItem.prototypes = { 
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}