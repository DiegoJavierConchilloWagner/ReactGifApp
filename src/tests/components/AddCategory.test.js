import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en <AddCategory />', () => {
    
    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCategories } /> );

    beforeEach ( () => {
        jest.clearAllMocks();
        wrapper = shallow( <AddCategory setCategories={ setCategories } /> );
    });

    test('Debe de mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });
    
    test('Debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo';

        input.simulate('change', { target: { value } });

        expect( wrapper.find('p').text().trim() ).toBe( value );
    });
    
    test('No debe de postear la informacion on submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).not.toHaveBeenCalled();
    });

    test('Debe de llamar el setCategories y limpiar la caja de texto', () => {

        const input = wrapper.find('input');
        const value = 'Hola Mundo';

        //Simular inputChange
        input.simulate('change', { target: { value } });

        //Simular el submit
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        //setCategories debe haber sido llamado
        expect( setCategories ).toHaveBeenCalled();
        //Debe haber sido llamado x cantidad veces
        expect( setCategories ).toHaveBeenCalledTimes(1);
        //Debe haber sido llamado por una function
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );

        //vaciar el input
        expect( wrapper.find('input').prop('value') ).toBe('');

    })
    
    
});
