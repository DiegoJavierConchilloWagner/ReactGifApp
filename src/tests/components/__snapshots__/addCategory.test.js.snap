// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> Debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
  </p>
  <input
    onChange={[Function]}
    placeholder="Buscar meme..."
    type="text"
    value=""
  />
</form>
`;
